test/test_searchlog.c

```diff
@@ -0,0 +1,152 @@
+#include "test.h"
+
+#include "../src/globalvars.h"
+
+#include "../src/tlf_curses.h"
+#include "../src/tlf_panel.h"
+#include "../src/searchlog.h"
+#include "../src/dxcc.h"
+
+// OBJECT ../src/searchlog.o
+// OBJECT ../src/zone_nr.o
+// OBJECT ../src/searchcallarray.o
+// OBJECT ../src/nicebox.o
+// OBJECT ../src/qtcutil.o
+// OBJECT ../src/printcall.o
+
+extern WINDOW *search_win;
+extern int nr_bands;
+
+const char *NOT_SET = "__NOT_SET__";
+
+static const char *msg, *msg1, *msg2;
+
+void showmsg(char *message) {
+    msg = message;
+}
+
+void showstring(char *message1, char *message2) {
+    msg1 = message1;
+    msg2 = message2;
+}
+
+/*********************/
+// mocks
+
+// stoptx.c
+int stoptx() {
+    return 0;
+}
+
+// ui_utils.c
+int modify_attr(int attr) {
+    return attr;
+}
+
+// get_time.c
+void get_time(void) {
+}
+
+// getpx.c
+void getpx(char *checkcall) {
+}
+
+// dxcc.c
+dxcc_data *dxcc_by_index(unsigned int index) {
+    return NULL; //oops
+}
+
+// getctydata.c
+int getctydata(char *checkcallptr) {
+    return 0;
+}
+/*********************/
+
+int setup_default(void **state) {
+    msg = msg1 = msg2 = NOT_SET;
+    arrlss = 0;
+    dxped = 0;
+    contest = 0;
+    search_win = NULL;
+    return 0;
+}
+
+// callmaster is checked first in current directory,
+// create it there
+static void write_callmaster(const char *content) {
+    FILE *f = fopen("callmaster", "w");
+    assert_non_null(f);
+    fputs(content, f);
+    fclose(f);
+}
+
+static void remove_callmaster() {
+    unlink("callmaster");   // no need to check if file exists
+}
+
+int teardown_default(void **state) {
+    remove_callmaster();
+    return 0;
+}
+
+void test_callmaster_no_file(void **state) {
+    remove_callmaster();
+    int n = load_callmaster();
+    assert_int_equal(n, 0);
```

Comment:
That test fails as soon as TLF is already installed. 'load_callmaster()' is also looking in the PACKAGE_DATA_DIR location where a callmaster file is present in that case.